flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -108,21 +111,38 @@ private static void mergeInto(JsonNode toNode, JsonNode fromNode) {
         }
     }
 
-    public static void deleteCluster(FlinkDeployment flinkApp, KubernetesClient kubernetesClient) {
+    public static void deleteCluster(
+            FlinkDeployment flinkApp,
+            KubernetesClient kubernetesClient,
+            boolean deleteHaConfigmaps) {
         deleteCluster(
                 flinkApp.getMetadata().getNamespace(),
                 flinkApp.getMetadata().getName(),
-                kubernetesClient);
+                kubernetesClient,
+                deleteHaConfigmaps);
     }
 
     public static void deleteCluster(
-            String namespace, String clusterId, KubernetesClient kubernetesClient) {
+            String namespace,
+            String clusterId,
+            KubernetesClient kubernetesClient,
+            boolean deleteHaConfigmaps) {
         kubernetesClient
                 .apps()
                 .deployments()
                 .inNamespace(namespace)
                 .withName(clusterId)
                 .cascading(true)
                 .delete();
+
+        if (deleteHaConfigmaps) {
```

Comment:
This seems only handle the situation for HA based on ConfigMap. Will this also work for the HA based on ZK ?

Reply:
You are absolutely correct, this only works for kubernetes HA (using configmaps). 

It's a good idea to add comments to the code and highlight in the README.

In the future we could support other HA providers but that is out of scope in this ticket.

Reply:
Make sense to me, I'm OK to merge the current shape with some more comments for clarification.  

But I have sense that we may have hard work to do the nice clean up work for other HA providers, It's a bit of out of scope of the operator responsibility or ability, maybe we should extend at the Flink to support `deleteAndCleanUpHA` ?. Do you have some inputs for this cc @wangyang0918

Reply:
The most appropriate way to the HA data clean up is `HighAvailabilityServices#closeAndCleanupAllData()`. It should work both for ZooKeeper and K8s HA. But I agree we could comment the limitation here and do the improvement later.